### PR TITLE
Remove dependency of apache-common-lang3 in the client.

### PR DESCRIPTION
**Pull request by zongtanghu**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

Remove dependency of apache-common-lang3 in the client, such as replacing NumberUtils with ConvertUtils in all modules of nacos.

## Brief changelog

The main goal is to remove dependency for apache commons-lang package.

## Verifying this change

I have already paased the unit test cases.

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by zongtanghu** [comment]
Hi @yanlinly @chuntaojun  please help to review this pr, thanks.

**Comment by yanlinly** [comment]
https://github.com/alibaba/nacos/pull/2952  is same?

**Comment by zongtanghu** [comment]
I have just removed the apache common-lang3 package in the client. @yanlinly

File: client/src/main/java/com/alibaba/nacos/client/naming/backups/FailoverReactor.java [hunk 1]
```diff
@@ -52,7 +59,7 @@ public FailoverReactor(HostReactor hostReactor, String cacheDir) {
         this.hostReactor = hostReactor;
         this.failoverDir = cacheDir + "/failover";
         // init executorService
-        this.executorService = Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
+        this.executorService = new ScheduledThreadPoolExecutor(1, new ThreadFactory() {
```
Comment (yanlinly) [inline client/src/main/java/com/alibaba/nacos/client/naming/backups/FailoverReactor.java:62]: 这里改的原因是？

Comment (zongtanghu) [reply client/src/main/java/com/alibaba/nacos/client/naming/backups/FailoverReactor.java:62]: 这个是PMD报出来的问题。。。 所以我就顺手修改了

File: common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java [hunk 2]
```diff
@@ -16,15 +16,17 @@
...
-import org.apache.commons.lang3.StringUtils;
-
 /**
  * @author <a href="mailto:liaochuntao@live.com">liaochuntao</a>
  */
 public final class ConvertUtils {
 
     private static final String NULL_STR = "null";
 
+    public static int toInt(String val) {
```
Comment (chuntaojun) [inline common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java:26]: You may need to add a comment, which automatically defaults to 0

File: common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java [hunk 3]
```diff
@@ -35,18 +37,164 @@ public static int toInt(String val, int defaultValue) {
...
+     *   BooleanUtils.toBooleanObject("Y")     = Boolean.TRUE // i.e. Y[ES]
+     *   BooleanUtils.toBooleanObject("blue")  = null
+     *   BooleanUtils.toBooleanObject("true ") = null // trailing space (too long)
+     *   BooleanUtils.toBooleanObject("ono")   = null // does not match on or no
+     * </pre>
+     *
+     * @param str  the String to check; upper and lower case are treated as the same
+     * @return the Boolean value of the string, {@code null} if no match or {@code null} input
+     */
+    @SuppressWarnings({"PMD.UndefineMagicConstantRule", "PMD.AvoidComplexConditionRule", "PMD.SwitchStatementRule"})
```
Comment (chuntaojun) [inline common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java:128]: 直接类上面，@SuppressWarnings（“all”）吧

File: common/src/main/java/com/alibaba/nacos/common/utils/CharSequenceUtils.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.nacos.common.utils;
+
+/**
+ * @author zongtanghu
+ */
+public class CharSequenceUtils {
```
Comment (chuntaojun) [inline common/src/main/java/com/alibaba/nacos/common/utils/CharSequenceUtils.java:21]: 不必要单独创建一个新的类，这个放到StringUtils内部就好了

File: common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java (hunk 2 above)
Comment (zongtanghu) [reply common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java:26]: Okay, I will add a comment in next commit.

File: common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java (hunk 3 above)
Comment (zongtanghu) [reply common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java:128]: Okay, I will adjust this codes in next commit.

**Comment by zongtanghu** [comment]
There is one ci test case failed, because it executed time out. @yanlinly @chuntaojun Can you help to rerun the ci job again?

**Comment by KomachiSion** [comment]
Please fix confilcts.

**Comment by zongtanghu** [comment]
> Please fix confilcts.

Okay, I will fix the codes' conflicts right now. @KomachiSion

**Comment by zongtanghu** [comment]
@yanlinly @KomachiSion @chuntaojun  I have already done the codes' conflicts.

File: client/src/main/java/com/alibaba/nacos/client/identify/STSConfig.java [hunk 5]
```diff
@@ -15,7 +15,8 @@
  */
 package com.alibaba.nacos.client.identify;
 
-import org.apache.commons.lang3.StringUtils;
+
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/identify/STSConfig.java:18]: remove extra blank line.

File: client/src/main/java/com/alibaba/nacos/client/logging/AbstractNacosLogging.java [hunk 6]
```diff
@@ -15,8 +15,9 @@
  */
 package com.alibaba.nacos.client.logging;
 
-import org.apache.commons.lang3.BooleanUtils;
-import org.apache.commons.lang3.StringUtils;
+
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/logging/AbstractNacosLogging.java:18]: remove extra blank line.

File: client/src/main/java/com/alibaba/nacos/client/naming/backups/FailoverReactor.java [hunk 7]
```diff
@@ -25,16 +25,23 @@
...
+import java.util.Map;
+import java.util.Date;
+import java.util.Calendar;
+import java.util.TimerTask;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.util.List;
 import java.util.concurrent.*;
 
+
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/naming/backups/FailoverReactor.java:44]: remove extra blank line.

File: common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java [hunk 8]
```diff
@@ -35,18 +44,164 @@ public static int toInt(String val, int defaultValue) {
...
+                    if ((ch0 == 'f' || ch0 == 'F') && (ch1 == 'a' || ch1 == 'A') && (ch2 == 'l' || ch2 == 'L') && (ch3 == 's' || ch3 == 'S') && (ch4 == 'e' || ch4 == 'E')) {
+                        return Boolean.FALSE;
+                    }
+            }
+
+            return null;
+        }
+    }
+
+    //   end
```
Comment (yanlinly) [inline common/src/main/java/com/alibaba/nacos/common/utils/ConvertUtils.java:205]: 这些代码是从哪里拷贝的，还是自己写的呢？ 如果是拷贝过来的注意一下license

File: common/src/main/java/com/alibaba/nacos/common/utils/StringUtils.java [hunk 9]
```diff
@@ -299,5 +299,84 @@ static boolean regionMatches(final CharSequence cs, final boolean ignoreCase, fi
...
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+    }
+
+    //   end
```
Comment (yanlinly) [inline common/src/main/java/com/alibaba/nacos/common/utils/StringUtils.java:381]: 这些代码是从哪里拷贝的，还是自己写的呢？ 如果是拷贝过来的注意一下license

File: client/src/main/java/com/alibaba/nacos/client/identify/STSConfig.java (hunk 5 above)
Comment (zongtanghu) [reply client/src/main/java/com/alibaba/nacos/client/identify/STSConfig.java:18]: In the lasted commit, the extra blank line has aleady been removed.

File: client/src/main/java/com/alibaba/nacos/client/naming/backups/FailoverReactor.java (hunk 7 above)
Comment (zongtanghu) [reply client/src/main/java/com/alibaba/nacos/client/naming/backups/FailoverReactor.java:44]: In the next commit, I will remove the blank line.

File: client/src/main/java/com/alibaba/nacos/client/logging/AbstractNacosLogging.java (hunk 6 above)
Comment (zongtanghu) [reply client/src/main/java/com/alibaba/nacos/client/logging/AbstractNacosLogging.java:18]: In the lasted commit, the extra blank line has aleady been removed.

**Pull request merged**